sdks/python/apache_beam/io/avroio_it_test.py

```diff
@@ -0,0 +1,237 @@
+    num_records = int(num_records) if num_records else 10000
+
+    # Seed a `PCollection` with indices that will each be FlatMap'd into
+    # `batch_size` records, to avoid having a too-large list in memory at
+    # the outset
+    batch_size = write_pipeline.get_option('batch-size')
+    batch_size = int(batch_size) if batch_size else 1000
+
+    assert batch_size < num_records
+    # pylint: disable=range-builtin-not-iterating
+    batches = range(int(num_records / batch_size))
+
+    def batch_indices(start):
+      # pylint: disable=range-builtin-not-iterating
+      return range(start * batch_size, (start + 1) * batch_size)
+
+    multiplier = 10
+    assert (multiplier % 2 == 0), \
+      "Use an even multiplier to not compare floats."
+
+    # A `PCollection` with `num_records` avro records
+    _ = (write_pipeline
+         | 'CreateBatches' >> Create(batches)
+         | 'ExpandBatches' >> FlatMap(batch_indices)
+         | 'Multiply' >> Map(lambda x: multiplier * x)
+         | 'CreateRecords' >> Map(record)
+         | 'WriteAvro' >> WriteToAvro(files_output,
+                                      self.SCHEMA,
+                                      file_name_suffix=".avro",
+                                      use_fastavro=self.use_fastavro)
+        )
+
+    write_result = write_pipeline.run()
+    write_result.wait_until_finish()
+    assert write_result.state == PipelineState.DONE
+
+    def check(x, mean):
+      if not x == mean:
+        raise BeamAssertException('Assertion failed: %s == %s' % (x, mean))
+
+    def check_distr(lst):
+      if not len(set(lst)) == 1:
+        raise BeamAssertException(
+            'Labels should be equally distributed: %s ' % (lst))
+
+    def check_concat(concatenated_string):
+      split_concatenated_string = concatenated_string.split(" ")
+      for i in range(len(batches) * batch_size):
+        if str(multiplier * i) not in split_concatenated_string:
+          raise BeamAssertException(
+              'String should be in concatenation: %s' % str(multiplier * i))
+
+    def check_for_none(lst):
+      if not None in lst:
+        raise BeamAssertException('None value was not preserved: %s ' % (lst))
+
+    def concat_strings(values):
+      return " ".join(values)
+
+    read_pcol = (read_pipeline
+                 | 'ReadAvro' >> ReadFromAvro(files_output + '*')
+                )
+
+    _ = (read_pcol
+         | 'ExtractLabel' >> Map(lambda x: x['label'])
+         | 'MakeLabelsKV' >> Map(lambda x: (x, 1))
+         | 'CountLabels' >> Count.PerKey()
+         | 'ExtractLabelCounts' >> Map(lambda x: x[1])
+         | 'LabelsToList' >> ToList()
+         | 'CheckEqualDistributionLabels' >> Map(lambda x: check_distr(x))
+        )
+
+    _ = (read_pcol
+         | 'ExtractNumber' >> Map(lambda x: x['number'])
+         | 'CalculateMean' >> Mean.Globally()
+         | 'MeanIsCorrect' >> Map(
+             lambda x: check(x, multiplier *
```

Comment:
Old comment valentyn: How do we compute expected value of Mean here?
Resolution: Pcollection is e.g. [(0 1 2) (3 4 5) (6 7 8) (9 10 11)] with batch_size = 3, batches = 4 with braces denoting the batches, so average is 3*4 - 1 = 5.5 . The purpose of multiplier is to avoid float comparison (resolved comment:  https://github.com/apache/beam/pull/8818/files#r292576662 ), so we first do the entire PCollection * an even multiplier, which results in the average being an integer.